master/internal/telemetry/reports.go

```diff
@@ -172,17 +172,17 @@ func fetchNumTrials(db db.DB, experimentID int) *int64 {
 	return &result
 }
 
-func fetchNumSteps(db db.DB, experimentID int) *int64 {
-	result, err := db.ExperimentNumSteps(experimentID)
+func fetchNumSteps(ctx context.Context, db db.DB, experimentID int) *int64 {
+	result, err := db.ExperimentNumSteps(ctx, experimentID)
```

Comment:
it you want to thread the context through that is fine with me. There is a lot of tech debt with not threading contexts in our codebase currently.

I would also be okay if you wanted to keep the change more focused. If you wanted to do that you could just add the ```context.TODO``` in this function so something like  

```suggestion
func fetchNumSteps(db db.DB, experimentID int) *int64 {
	result, err := db.ExperimentNumSteps(context.TODO(), experimentID)
```

Reply:
I would prefer having changes more focused to bunify and test coverage. So for now I have added `context.TODO()` as you suggested.
We may complete the `context.TODO()` in another PR.